server/command.go

```diff
@@ -147,31 +164,147 @@ func (p *Plugin) executeCommandQueue(args *model.CommandArgs) *model.CommandResp
+	}
 
+	_, appErr := p.API.CreatePost(&model.Post{
+		UserId:    args.UserId,
+		ChannelId: args.ChannelId,
+		RootId:    args.RootId,
+		Message:   fmt.Sprintf("#### %v %v) %v", hashtag, numQueueItems, message),
+	})
 	if appErr != nil {
-		return responsef("Error calculating list number")
+		return responsef("Error creating post: %s", appErr.Message)
+	}
+
+	return &model.CommandResponse{}
+}
+
+func calculateQueItemNumberAndUpdateOldItems(meeting *Meeting, args *model.CommandArgs, p *Plugin, hashtag string) (int, *model.CommandResponse) {
+	searchResults, appErr := p.API.SearchPostsInTeamForUser(args.TeamId, args.UserId, model.SearchParameter{Terms: &hashtag})
+	if appErr != nil {
+		return 0, responsef("Error calculating list number")
+	}
+
+	counter := 1
+
+	var sortedPosts []*model.Post
+	// TODO we won't need to do this once we fix https://github.com/mattermost/mattermost-server/issues/11006
+	for _, post := range searchResults.PostList.Posts {
+		sortedPosts = append(sortedPosts, post)
+	}
+
+	sort.Slice(sortedPosts, func(i, j int) bool {
+		return sortedPosts[i].CreateAt < sortedPosts[j].CreateAt
+	})
+
+	for _, post := range sortedPosts {
+		_, parsedMessage, _ := parseMeetingPost(meeting, post)
+		_, findErr := p.API.UpdatePost(&model.Post{
+			Id:        post.Id,
+			UserId:    args.UserId,
+			ChannelId: args.ChannelId,
+			RootId:    args.RootId,
+			Message:   fmt.Sprintf("#### %v %v) %v", hashtag, counter, parsedMessage.textMessage),
+		})
+		counter++
+		if findErr != nil {
+			return 0, responsef("Error updating post: %s", findErr.Message)
+		}
+	}
+	return counter, nil
+}
+
+func (p *Plugin) executeCommandReQueue(args *model.CommandArgs) *model.CommandResponse {
+	split := strings.Fields(args.Command)
+
+	if len(split) <= 2 {
+		return responsef("Missing parameters for requeue command")
+	}
+
+	meeting, err := p.GetMeeting(args.ChannelId)
+	if err != nil {
+		return responsef("Can't find the meeting")
 	}
 
-	postList := *searchResults.PostList
-	numQueueItems := len(postList.Posts)
+	oldPostID := split[2]
+	postToBeReQueued, _ := p.API.GetPost(oldPostID)
+	hashtagDateFormat, parsedMeetingMessage, errors := parseMeetingPost(meeting, postToBeReQueued)
+	if errors != nil {
+		return errors
+	}
+
+	originalPostDate := strings.ReplaceAll(strings.TrimSpace(parsedMeetingMessage.date), "_", " ") // reverse what we do to make it a valid hashtag
+	originalPostMessage := strings.TrimSpace(parsedMeetingMessage.textMessage)
+
+	today := time.Now()
+	local, _ := time.LoadLocation("Local")
+	formattedDate, _ := time.ParseInLocation(hashtagDateFormat, originalPostDate, local)
+	if formattedDate.Year() == 0 {
+		thisYear := today.Year()
+		formattedDate = formattedDate.AddDate(thisYear, 0, 0)
+	}
 
-	_, appErr = p.API.CreatePost(&model.Post{
+	if today.Year() <= formattedDate.Year() && today.YearDay() < formattedDate.YearDay() {
+		return responsef("We don't support re-queuing future items, only available for present and past items.")
```

Comment:
return responsef("We don't support re-queuing future items, only available for existing items.")